public/js/model/game.js

```diff
@@ -0,0 +1,145 @@
+      this.scoreCard[i] = this.rollHistory[i][0] + this.rollHistory[i][1];
+    } else if (this.rollHistory[i][0] === 10) {
+      this.scoreCard[i] = this.rollHistory[i][0];
+    };
+  };
+  for (var i = 1; i <= 9; i++) {
+    var nextTwoFrames = this.rollHistory[i + 1].concat(this.rollHistory[i + 2]);
+    if (this.rollHistory[i][0] === 10 && (nextTwoFrames[0] + nextTwoFrames[1])) {
+      this.scoreCard[i] += (nextTwoFrames[0] + nextTwoFrames[1]);
+    } else if (this.rollHistory[i][0] + this.rollHistory[i][1] === 10 && nextTwoFrames[0]) {
+      this.scoreCard[i] += nextTwoFrames[0]
+    };
+  };
+  if (this.rollHistory[10][0] === 10 && this.rollHistory[11][1]) {
+    this.scoreCard[10] += (this.rollHistory[11][0] + this.rollHistory[11][1]);
+  } else if (this.rollHistory[10][0] + this.rollHistory[10][1] === 10 && this.rollHistory[11][0]) {
+    this.scoreCard[10] += this.rollHistory[11][0];
+  };
+  return this.scoreCard
+};
+
+Game.prototype.accumulate = function () {
+  var scoreAccumulation = 0
+  for (var i = 1; i <= 10; i++) {
+    scoreAccumulation += this.scoreCard[i];
+    if (this.scoreCard[i] != "") {
+      this.cumulativeScore[i] = ' : ' + scoreAccumulation;
+    };
+  };
+  return this.cumulativeScore;
+};
+
+Game.prototype._isNormalFrame = function (frame) {
+  return (this._isIncompleteFrame(frame) &&
+          this._isNotStrikeFrame(frame) &&
+          this._isNotBonusFrame(frame));
+};
+
+Game.prototype._isBonusRoll = function (frame) {
+  return (this._isEleventhFrame(frame) &&
+          this._isFollowingTwoRollFrame(frame) &&
+          this._isFollowingSpare(frame) &&
+          this._isEmptyFrame(frame));
+};
+
+Game.prototype._isBonusFrame = function (frame) {
+  return (this._isEleventhFrame(frame) &&
+          this._isFollowingStrike(frame) &&
+          this._isIncompleteFrame(frame));
+};
+
+Game.prototype._isIncompleteFrame = function (frame) {
+  return (this.rollHistory[frame].length < 2);
+};
+
+Game.prototype._isNotStrikeFrame = function (frame) {
+  return (this.rollHistory[frame][0] != 10);
+};
+
+Game.prototype._isNotBonusFrame = function (frame) {
+  return (frame != 11);
+};
+
+Game.prototype._isEleventhFrame = function (frame) {
+  return (frame === 11);
+};
+
+Game.prototype._isFollowingTwoRollFrame = function (frame) {
+  return (this.rollHistory[frame - 1].length === 2);
+};
+
+Game.prototype._isFollowingSpare = function (frame) {
+  return ((this.rollHistory[frame - 1][0] + this.rollHistory[frame - 1][1]) === 10);
+};
+
+Game.prototype._isFollowingStrike = function (frame) {
+  return (this.rollHistory[frame - 1][0] === 10);
+};
+
+Game.prototype._isEmptyFrame = function (frame) {
+  return (this.rollHistory[frame].length === 0);
+};
+
+Game.prototype._isLegalRoll = function (frame, score) {
+  if (this._isEmptyFrame(frame)) {
+    return true;
+  } else {
+    return (this.rollHistory[frame][0] + score <= 10);
+  };
+};
+
+Game.prototype._randomInsult = function () {
```

Comment:
now this is pretty neat - I liked the way you used math.random to give some nice insults :) This showed creativity and your ability to think outside the box and think about the end-user experience in your app :)